### ISSUE TITLE
floats和模型参数量

### ISSUE THREAD

**Issue by jieduring** [luow23/INP-Former#33]
根据论文中使用的模型和图片大小，输出的floats和params与论文中相差很大，怎么回事呢

**Comment by jieduring** [comment]
[

![Image](https://github.com/user-attachments/assets/51d1ea45-215c-4179-a424-4abc9544b4be)

](url)

**Comment by jieduring** [comment]
![Image](https://github.com/user-attachments/assets/d2cfcde2-2ffa-4bc4-baa0-09520a3be821)
在验证前加入了FlopsProfiler类计算的

**Comment by luow23** [comment]
这个FLOPs可能是因为你输入的是一个batch，所以FLOPs会很大

**Comment by luow23** [comment]
这是我们计算参数量和FLOPs的代码
model = model.to(device)
    dummy_input = torch.randn(1, 3, 392, 392, dtype=torch.float).to(device)
    model.eval()
    flops, params = profile(model, (dummy_input,))
    print('flops: ', flops, 'params: ', params)
    print('flops: %.2f G, params: %.2f M' % (flops / 1000000000.0, params / 1000000.0))

**Comment by jieduring** [comment]
![Image](https://github.com/user-attachments/assets/12e236cf-25aa-46c6-8987-a61142c8f012)
用的vit-base，按照给的输入，FPS和参数量还是很高

**Comment by luow23** [comment]
这是我们跑出的结果
![Image](https://github.com/user-attachments/assets/7475616a-8793-40c6-81ae-b4ed74b731cd)

以下是我们的测试代码

![Image](https://github.com/user-attachments/assets/3df49319-c2f8-4539-9fce-c692d82757e0)

**Comment by luow23** [comment]
另外,我用我们这种计算FLOPs的方式计算了一下Dinomaly的FLOPs，发现FLOPs和其论文汇报的结果一样。

**Comment by jieduring** [comment]
可以给一下计算的脚本吗，试了好几种，params都是155.10，flops也很难达到论文中的数值。

**Comment by luow23** [comment]
没问题，我们提供我们计算的脚本，你需要安装thop库，from thop import profile。args和主实验里面的一样即可。
```
def test_FPS(args):
    # Fixing the Random Seed
    setup_seed(1)

    # Adopting a grouping-based reconstruction strategy similar to Dinomaly
    target_layers = [2, 3, 4, 5, 6, 7, 8, 9]
    fuse_layer_encoder = [[0, 1, 2, 3], [4, 5, 6, 7]]
    fuse_layer_decoder = [[0, 1, 2, 3], [4, 5, 6, 7]]

    # Encoder info
    encoder = vit_encoder.load(args.encoder)
    if 'small' in args.encoder:
        embed_dim, num_heads = 384, 6
    elif 'base' in args.encoder:
        embed_dim, num_heads = 768, 12
    elif 'large' in args.encoder:
        embed_dim, num_heads = 1024, 16
        target_layers = [4, 6, 8, 10, 12, 14, 16, 18]
    else:
        raise "Architecture not in small, base, large."

    # Model Preparation
    Bottleneck = []
    INP_Guided_Decoder = []
    INP_Extractor = []

    # bottleneck
    Bottleneck.append(Mlp(embed_dim, embed_dim * 4, embed_dim, drop=0.))
    Bottleneck = nn.ModuleList(Bottleneck)

    # INP
    INP = nn.ParameterList(
                    [nn.Parameter(torch.randn(args.INP_num, embed_dim))
                     for _ in range(1)])

    # INP Extractor
    for i in range(1):
        blk = Aggregation_Block(dim=embed_dim, num_heads=num_heads, mlp_ratio=4.,
                                qkv_bias=True, norm_layer=partial(nn.LayerNorm, eps=1e-8))
        INP_Extractor.append(blk)
    INP_Extractor = nn.ModuleList(INP_Extractor)

    # INP_Guided_Decoder
    for i in range(8):
        blk = Prototype_Block(dim=embed_dim, num_heads=num_heads, mlp_ratio=4.,
                              qkv_bias=True, norm_layer=partial(nn.LayerNorm, eps=1e-8))
        INP_Guided_Decoder.append(blk)
    INP_Guided_Decoder = nn.ModuleList(INP_Guided_Decoder)

    model = INP_Former(encoder=encoder, bottleneck=Bottleneck, aggregation=INP_Extractor, decoder=INP_Guided_Decoder,
                             target_layers=target_layers,  remove_class_token=True, fuse_layer_encoder=fuse_layer_encoder,
                             fuse_layer_decoder=fuse_layer_decoder, prototype_token=INP)

    # Test FLOPs and Params
    model = model.to(device)
    dummy_input = torch.randn(1, 3, 392, 392, dtype=torch.float).to(device)
    flops, params = profile(model, (dummy_input, ))
    print('flops: ', flops, 'params: ', params)
    print('flops: %.2f G, params: %.2f M' % (flops / 1000000000.0, params / 1000000.0))

**Comment by jieduring** [comment]
感谢感谢！
